enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/TransactionLogCatchUpWriter.java

```diff
@@ -52,14 +53,19 @@
     private long lastTxId = -1;
     private long expectedTxId;
 
-    TransactionLogCatchUpWriter( File storeDir, FileSystemAbstraction fs, PageCache pageCache,
-            LogProvider logProvider, long fromTxId, boolean asPartOfStoreCopy ) throws IOException
+    TransactionLogCatchUpWriter( File storeDir, FileSystemAbstraction fs, PageCache pageCache, Config config,
+            LogProvider logProvider, long fromTxId, boolean asPartOfStoreCopy, boolean keepTxLogsInStoreDir ) throws IOException
```

Comment:
Would it be enough to only let the config own this decision, or do we explicitly need this `keepTxLogsInStoreDir` flag?

Reply:
we need to set it explicitly since it can be the case that now we do whole store copy that will do transaction pulling in the end and we can't use the same shared config since its hidden under RemoteStore abstraction which is the same for case when we do whole store copy or only transaction pulling (in first case we need to put logs into correct configured place and in the second into the same directory)

Reply:
OK gotcha